File: src/main/java/com/home/domain/DataRepository.java
package com.home.domain;

import com.home.dto.Data;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class DataRepository {

    private Map<Long, Data> store = new ConcurrentHashMap<>();

    public DataRepository() {
        store.put(1L, new Data(1L, "first"));
        store.put(2L, new Data(2L, "second"));
    }

    public void update(Data data) {
        create(data);
    }

    public void create(Data data) {
        store.put(data.getId(), data);
    }

    public Data read(Long id) {
        return store.get(id);
    }

    public boolean delete(Long id) {
        return store.remove(id) != null;
    }
}

File: src/main/kotlin/com/home/domain/DataRepository.kt
package com.home.domain

import com.home.dto.Data
import java.util.*
import java.util.concurrent.ConcurrentHashMap

object DataRepository {
    private val store: MutableMap<Long, Data> = ConcurrentHashMap(hashMapOf(
        1L to Data(1L, "first"),
        2L to Data(2L, "second")
    ))

    fun update(data: Data): Unit {
        store.put(data.id, data)
    }

    fun create(data: Data) {
        val r = Random()
        data.id = r.nextLong()
        this.update(data)
    }

    fun read(id: Long): Data? = store.get(id)

    fun delete(id: Long): Data? = store.remove(id)
}
